### ISSUE TITLE
anything is wrong with serial config

### ISSUE THREAD

**Issue by TheIsotopes** [betaflight/betaflight-configurator#1864]
since latest configurator build the serial config have any issue

for example, i change the name of the quad and after comparing the diff this comes out:

left original and right after change the name
![compared](https://user-images.githubusercontent.com/29702171/73133369-f665f280-4027-11ea-98b4-22db282f4994.jpg)

after looking through all the tabs, i also get a repeat of six readouts in the ports tab

![ports_tab](https://user-images.githubusercontent.com/29702171/73133435-ce2ac380-4028-11ea-95c1-017e5bd8bb65.jpg)

I do not know if it is an configurator or a firmware issue

@fiam pls can you check this

**Comment by mikeller** [comment]
Yop, got a reproduction:
- flash OMNIBUSF4 Unified Target with STM32F405 4.2.0 from `master` with configurator 10.7.0 from `master`;
- edit any setting in the Configuration tab;
- Save & Reboot.

Expected behaviour:
- the Ports tab shows 4 ports.

Actual behavior:
- the Ports tab shows 8 ports:

![image](https://user-images.githubusercontent.com/4742747/73133636-863f8700-4090-11ea-97e2-73dfa59a4925.png)

**Comment by fiam** [comment]
Just found the problem for the duplicated ports. I’m working on the other problem right now and I’ll submit a PR in a bit.

**Comment by mikeller** [comment]
First problem that I found:

```
mikeller@pc:~/git/betaflight-configurator$ git diff
diff --git a/src/js/msp/MSPHelper.js b/src/js/msp/MSPHelper.js
index e3d82d6d..a3380711 100644
--- a/src/js/msp/MSPHelper.js
+++ b/src/js/msp/MSPHelper.js
@@ -1896,7 +1896,6 @@ MspHelper.prototype.crunch = function(code) {
 
                 const functionMask = self.serialPortFunctionsToMask(serialPort.functions);
                 buffer.push32(functionMask)
-                    .push8(self.BAUD_RATES.indexOf(serialPort.msp_baudrate))
                     .push8(self.BAUD_RATES.indexOf(serialPort.msp_baudrate))
                     .push8(self.BAUD_RATES.indexOf(serialPort.gps_baudrate))
                     .push8(self.BAUD_RATES.indexOf(serialPort.telemetry_baudrate))
```

Not sure if it's the same that you found, but I'll hold off with more investigation until you let me know what you have.

**Comment by fiam** [comment]
@mikeller That was the problem that was incorrectly setting the baudrates. I duplicated that line to test if the FC coped correctly with a payload longer than expected and then forgot to delete it.

The duplicated ports were caused because `SERIAL_CONFIG.ports` needed to be explicitly emptied, like in the handler for `MSP_CF_SERIAL_CONFIG`. I've just submitted fixed for both in #1865.
